native/src/modules/app/components/App.js

```diff
@@ -15,11 +15,31 @@ import NavigatorContainer from '../containers/NavigatorContainer'
 import { SafeAreaProvider } from 'react-native-safe-area-context'
 import StaticServerProvider from '../../static-server/containers/StaticServerProvider'
 import { NavigationContainer } from '@react-navigation/native'
+import PermissionSnackbarContainer from '../../layout/containers/PermissionSnackbarContainer'
 
 const dataContainer: DataContainer = new DefaultDataContainer()
 const store: Store<StateType, StoreActionType> = createReduxStore(dataContainer)
 
 const App = () => {
+  const [routeName, setRouteName] = useState<?string>(null)
+  const [routeKey, setRouteKey] = useState<?string>(null)
+  const [cityCode, setCityCode] = useState<?string>(null)
+  const [languageCode, setLanguageCode] = useState<?string>(null)
+
+  const onStateChange = useCallback(state => {
+    if (state) {
+      const route = state.routes[state.index]
+      setRouteName(route.name)
+      setRouteKey(route.key)
+      if (route.params?.cityCode && route.params?.languageCode) {
+        // $FlowFixMe cityCode is not of type mixed
```

Comment:
Is there a way of type the parameters? That is possible with typescript at least :sweat:

Reply:
Yes, I know. Did not work for me at least.

Reply:
Nice, good idea. Thanks!